bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NAEnergyModule.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.netatmo.internal.api.dto;
+
+/**
+ *
+ * @author Markus Dillmann - Initial contribution
+ *
+ */
+
+public class NAEnergyModule extends NADevice {
+    private int firmware_revision = -1;
+    private int rf_strength;
+    private String battery_state = "";
+
+    public String getBatteryState() {
+        return this.battery_state;
+    }
+
+    public void setBatteryState(String battery_state) {
```

Comment:
Actually I think it is needed currently ...
Bernhard introduced the additional deserialisation objects and I extended the code he used for setting NARoom properties to handle NRV and NAPlug in the HomeEnergyHandler...
Ultimately I wanted to have all needed info for the channels in the NAThing/Device hierarchy so that the handlers behave the same.
I know this is not optimal, but I wanted to have static info from "homesdata" like setup_date in the same NAModule instance as dynamic data from "homestatus" like battery_level (and firmware) so that I only need to handle one in the updatesettings&properties.
Any ideas welcome on how to handle this.

Reply:
Try to stick with current DTO object hierarchy and overload it appropriatly (NAObject -> NAThing -> NAModule -> NASpecificWhathever), this will help keep the code as simple as possible.